Tella/Data/VaultManager/VaultFilesManagerExtension.swift

```diff
@@ -0,0 +1,148 @@
+//
+//  VaultFilesManagerExtension.swift
+//  Tella
+//
+//  Created by Dhekra Rouatbi on 13/6/2024.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+import Foundation
+import MobileCoreServices
+import Photos
+
+extension VaultFilesManager {
+    
+    func updateURL(importedFile:inout ImportedFile) async {
+        
+        do {
+            
+            var url : URL?
+            guard let asset = importedFile.asset else {
+                return
+            }
+            
+            let mediaType = asset.mediaType
+            
+            if (importedFile.urlFile != nil && mediaType == .image) {
+                return // Image data already saved into the temp fileURL
+            }
+            
+            switch asset.mediaType {
+            case .image:
+                url = try await getImageUrlFromAsset(importedFile:importedFile)
+            default:
+                url = try await asset.getAVAssetUrl()
+            }
+            
+            importedFile.urlFile = url
+            
+        } catch let error {
+            debugLog(error)
+            return
+        }
+    }
+    
+    func getImageUrlFromAsset(importedFile:ImportedFile) async throws -> URL? {
+        
+        guard let asset = importedFile.asset else {
+            throw RuntimeError("Asset is nil")
+        }
+        
+        var data = try await asset.getDataFromAsset()
+        
+        if !(importedFile.shouldPreserveMetadata) {
+            guard let newData = data?.byRemovingEXIF() else { return nil}
+            data = newData
+        }
+        
+        guard let resource = asset.getAssetResource() else {
+            throw RuntimeError("Error fetching asset resources")
+        }
+        
+        let fileURL =  self.vaultManager?.saveDataToTempFile(data: data,
+                                                             fileName: resource.originalFilename)
+        
+        return fileURL
+    }
+    
+    func getModifiedURL(importedFile:ImportedFile) async -> URL? {
+        
+        var filePath : URL?
+        
+        if let _ = importedFile.asset {
+            await filePath = getUrlFromAsset(importedFile: importedFile)
+            
+        } else {
+            await filePath = getModifiedUrlFromURL(importedFile: importedFile)
+        }
+        
+        return filePath
+    }
+    
+    func getModifiedUrlFromURL(importedFile:ImportedFile) async -> URL? {
+        
+        guard let urlFile = importedFile.urlFile else { return nil}
+        
+        if importedFile.shouldPreserveMetadata {
+            return urlFile
+        } else {
+            let type = urlFile.fileType
```

Comment:
can we add this whole block into a new private function?

Reply:
I also added `private`  to the other functions